### DISCUSSION TITLE
10 instances of Prisma Client actively running.

### DISCUSSION THREAD

**Discussion by devpedrodiass** [prisma/prisma#23991] [Q&A]
### The Context
I'm facing right now this problem:
> warn(prisma-client) There are already 10 instances of Prisma Client actively running.

But it is not in `Development`, I'm facing that in the Production. I'm using Vercel to Deploy.

This is my `db.ts` that I call every time I need to go to the database

```ts
import { PrismaClient } from "@prisma/client";

const prismaClientSingleton = () => {
  return new PrismaClient();
};

declare global {
  var prismaGlobal: undefined | ReturnType<typeof prismaClientSingleton>;
}

console.log(
  "---> PRISMA - Instancia Existente: ",
  globalThis.prismaGlobal ? true : false,
  " - NODE_ENV: ",
  process.env.NODE_ENV
);

export const db = globalThis.prismaGlobal ?? prismaClientSingleton();

if (process.env.NODE_ENV !== "production") {
  console.log("---> PRISMA - Instancia DEV");
  globalThis.prismaGlobal = db;
}
```

How can I fix that?

### Information about Prisma Schema, Client Queries and Environment (optional)

```json
{
 "@prisma/client": "^5.11.0",
 "next": "14.1.3",
 "prisma": "^5.11.0"
}

```

**Comment by ludralph** [comment]
Hi @devpedrodiass 👋 

Can you please try the suggestion in this [comment](https://github.com/prisma/prisma/discussions/4399#discussioncomment-9107335). Update your code as stated in the comment and let us know if the issue persist

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
